pkg/scheduler/watchers/membercluster/controller.go

```diff
@@ -190,20 +192,73 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 				return false
 			}
 
-			// Capture label changes.
-			//
 			clusterKObj := klog.KObj(newCluster)
 			// The cluster is being deleted.
 			if oldCluster.GetDeletionTimestamp().IsZero() && !newCluster.GetDeletionTimestamp().IsZero() {
 				klog.V(2).InfoS("A member cluster is leaving the fleet", "memberCluster", clusterKObj)
 				return true
 			}
+
+			// Capture label changes.
+			//
 			// Note that the controller runs only when label changes happen on joined clusters.
 			if !reflect.DeepEqual(oldCluster.Labels, newCluster.Labels) {
 				klog.V(2).InfoS("A member cluster label change has been detected", "memberCluster", clusterKObj)
 				return true
 			}
 
+			// Capture non-resource property changes.
+			//
+			// Observation time refreshes is not considered as a change.
+			oldProperties := oldCluster.Status.Properties
+			newProperties := newCluster.Status.Properties
+			if len(oldProperties) != len(newProperties) {
+				return true
+			}
+			for oldK, oldV := range oldProperties {
+				newV, ok := newProperties[oldK]
+				if !ok || oldV.Value != newV.Value {
+					return true
+				}
+			}
+
+			// Capture resource usage changes.
+			oldCapacity := oldCluster.Status.ResourceUsage.Capacity
+			newCapacity := newCluster.Status.ResourceUsage.Capacity
+			if len(oldCapacity) != len(newCapacity) {
+				return true
+			}
+			for oldK, oldV := range oldCapacity {
+				newV, ok := newCapacity[oldK]
+				if !ok || !oldV.Equal(newV) {
+					return true
+				}
+			}
```

Comment:
ditto

Reply:
This one can actually use the deepequal methods, but we still have to compare them separately.